src/metabase/models/setting.clj

```diff
@@ -1081,20 +1081,22 @@
   []
   (str/ends-with? (ns-name *ns*) "-test"))
 
-(defn- deref-map [form]
+(defn- deref-map [expr]
   (cond
-    (map? form)    form
-    (symbol? form) (u/prog1 @(resolve form) (assert (map? <>)))
-    :else          (throw (ex-info "Unexpected :base in setting configuration. Expected a literal map, or var one."
-                                   {:form form}))))
-
-(defn- inject-base
-  "Recursively uses the values in `:base` as default values for `setting-options`"
-  [{:keys [base] :as setting-options}]
-  (let [without-base (dissoc setting-options :base)]
-    (if-not base
-      without-base
-      (merge (inject-base (deref-map base)) without-base))))
+    (map? expr)    expr
+    (symbol? expr) (u/prog1 @(resolve expr)
+                     (when-not (map? <>)
+                       (throw (ex-info "Invalid :base in setting configuration. Must reference a map."
+                                       {:symbol expr :value <>})))
+                     (assert (map? <>)))
+    :else          (throw (ex-info "Invalid :base in setting configuration. Must be a literal map or a reference."
+                                   {:expression expr}))))
```

Comment:
So, you can't do something like 

```clj
(defsetting my-setting
  "My Setting"
  (merge map-1 map-2 {:a 100})
```

That seems user-unfriendly, I don't think you would expect it to work like that. I think we should rework the macro so it's not checking if its args are symbols and calling resolve on them... have the macro just serve as syntactic sugar and do the `def` and then have it do the important stuff in a function call in the macroexpansion

Reply:
I really like the way we structure our macros to delegate to runtime function, and hadn't looked into why we we're going that route in this case. It turned out there were a few reasons, but with some experimentation I was able to find my way around them - I've cut a [standalone PR](https://github.com/metabase/metabase/pull/38128) to do this, and rebased this one so it doesn't have to fuss around following references or leaving gotchas like this.

Reply:
In practice I don't really like the ergonomics of explicit `assoc` and `merge` calls with this macro, I feel that we can be more opinionated and have a bit of sugar for the pattern I found myself wanting to use - inheriting from single "base type". I thought that a trailing "base" map was an elegant and minimal solution, but now I realize should *override* the key value literals to match Clojure semantics I'm going to fall back to some of the other ideas I had for that mechanism. 

While your example would be nice for a "multiple mixin" scenario, I'm not sure we have many use cases for that, and even so I would suspect these would not be "once off" combinations, and we might as well def the recipe rather than inline to construction. I'm still very wet behind the ears and haven't used settings in anger much, so perhaps there are good cases I just haven't thought of.